kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt

```diff
@@ -263,35 +271,44 @@ class NotificationManager internal constructor(private val context: Context, pri
     override fun onNotificationPosted(notificationId: Int, notification: Notification, ongoing: Boolean) {
         scope.launch {
             event.updateNotificationState(NotificationState.POSTED(notificationId, notification, ongoing))
+
+            reload()
         }
     }
 
     override fun onNotificationCancelled(notificationId: Int, dismissedByUser: Boolean) {
         scope.launch {
             event.updateNotificationState(NotificationState.CANCELLED(notificationId))
+
+            reload()
         }
     }
 
-    internal fun onPlay() {
+    internal fun onPlay() = scope.launch {
         mediaSession.isActive = true
         reload()
     }
 
-    internal fun onPause() {
+    internal fun onPause() = scope.launch {
         reload()
     }
 
-    internal fun destroy() {
-        descriptionAdapter?.release()
+    fun onUnbind() = scope.launch {
+        reload()
+    }
+
+    internal fun destroy() = scope.launch {
         internalManager?.setPlayer(null)
+        mediaSession.isActive = false
+        descriptionAdapter.release()
     }
```

Comment:
The `clearNotification contents were transferred here, which I believe are a more suitable place.